### ISSUE TITLE
Problem with links

### ISSUE THREAD

**Issue by AntonSmatanik** [thephpleague/commonmark#278]
I don't know why but it is not converting links if they are in `<div>` or `<p>` elements. Example:

`<div> <p> You can get into contact with Galaxy Pig via e-mail send to [support@galaxypig.com](mailto:support@galaxypig.com) or via our Live chat, which is available in the time between on working days 10.00 and 22.00 o´clock CET. </p></div>`

Should be:

`<div> <p> You can get into contact with Galaxy Pig via e-mail send to <a href="mailto:support@galaxypig.com">support@galaxypig.com</a> or via our Live chat, which is available in the time between on working days 10.00 and 22.00 o´clock CET. </p></div>`

**Comment by colinodell** [comment]
Per the CommonMark spec, your input is parsed as a [type-6 HTML block](http://spec.commonmark.org/0.27/#html-blocks) and not an HTML inline.

> Everything until the next blank line or end of document gets included in the HTML block. So, in the following example, what looks like a Markdown code block is actually part of the HTML block, which continues until a blank line or the end of the document is reached:

You can confirm this behavior by [testing against the reference implementation here](http://spec.commonmark.org/dingus/?text=%3Cdiv%3E%20%3Cp%3E%20You%20can%20get%20into%20contact%20with%20Galaxy%20Pig%20via%20e-mail%20send%20to%20%5Bsupport%40galaxypig.com%5D(mailto%3Asupport%40galaxypig.com)%20or%20via%20our%20Live%20chat%2C%20which%20is%20available%20in%20the%20time%20between%20on%20working%20days%2010.00%20and%2022.00%20o%C2%B4clock%20CET.%20%3C%2Fp%3E%3C%2Fdiv%3E)

**Comment by AntonSmatanik** [comment]
So what can I do about it?

**Comment by colinodell** [comment]
Make the following changes to your input:

1. Place empty lines between the HTML tags and the paragraph content
2. Remove the `<p>` tags since those will always be added to paragraph content

For example:

```
<div>

You can get into contact with Galaxy Pig via e-mail send to [support@galaxypig.com](mailto:support@galaxypig.com) or via our Live chat, which is available in the time between on working days 10.00 and 22.00 o´clock CET.

</div>
```

This will give you the desired output according to the CommonMark specification.

**Comment by AntonSmatanik** [comment]
And when p element has some important tags? Like I'm parsing mixed html page into markdown and p element has some data tags. So I have decided to keep it like it is. Now I'm converting back from markdown to html but...
